lib/pluginUtils.js

```diff
@@ -0,0 +1,40 @@
+'use strict'
+
+const assert = require('assert')
+
+function getMeta (fn) {
+  return fn[Symbol.for('plugin-meta')]
+}
+
+function shouldSkipOverride (fn) {
+  return !!fn[Symbol.for('skip-override')]
+}
+
+function checkDependencies (fn) {
+  const meta = getMeta(fn)
+  if (!meta) return
+
+  const deps = meta.dependencies
+  if (!deps) return
+
+  if (deps.fastify) _checkDependencies.call(this, 'Fastify', deps.fastify)
+  if (deps.reply) _checkDependencies.call(this._Reply, 'Reply', deps.reply)
+  if (deps.request) _checkDependencies.call(this._Request, 'Request', deps.request)
+}
+
+function _checkDependencies (instance, deps) {
+  assert(Array.isArray(deps), 'The dependencies should be an array of strings')
```

Comment:
Do we not want to be complete and support more than just a plugin name? Maybe something like `{name: 'foo', version: 'semver range'}` instead of just a string?

Reply:
Yes, we can do that, but how the plugin developer will declare the plugin version?
We are already using the `version` key to check the installed fastify version.

Reply:
Change the signature: `fp(plugin, fastifyVersionRequirement, options)`.